qiskit/quantum_info/synthesis/two_qubit_decompose.py

```diff
@@ -1402,4 +1403,47 @@ def num_basis_gates(self, unitary):
         return np.argmax([trace_to_fid(traces[i]) * self.basis_fidelity ** i for i in range(4)])
 
 
-two_qubit_cnot_decompose = TwoQubitBasisDecomposer(CXGate())
+# This weird duplicated lazy structure is for backwards compatibility; Qiskit has historically
+# always made ``two_qubit_cnot_decompose`` available publicly immediately on import, but it's quite
+# expensive to construct, and we want to defer the obejct's creation until it's actually used.  We
+# only need to pass through the public methods that take `self` as a parameter.  Using `__getattr__`
+# doesn't work because it is only called if the normal resolution methods fail.  Using
+# `__getattribute__` is too messy for a simple one-off use object.
```

Comment:
Oh, too bad I was excited when I opened #7498 that we'd have the first use case for module level getattr now that we're >=3.7 but I guess not. Having this comment here is useful for explaining this I should have added one for the lazy aer and ibmq wrappers